pkg/kubectl/sorter.go

```diff
@@ -316,3 +305,76 @@ func (r *RuntimeSort) OriginalPosition(ix int) int {
 	}
 	return r.origPosition[ix]
 }
+
+type TableSorter struct {
+	field string
+	obj   *metav1beta1.Table
+}
+
+func (t *TableSorter) Len() int {
+	return len(t.obj.Rows)
+}
+
+func (t *TableSorter) Swap(i, j int) {
+	t.obj.Rows[i], t.obj.Rows[j] = t.obj.Rows[j], t.obj.Rows[i]
+}
+
+func (t *TableSorter) Less(i, j int) bool {
+	iObj := t.obj.Rows[i].Object.Object
+	jObj := t.obj.Rows[j].Object.Object
+
+	var iValues [][]reflect.Value
+	var jValues [][]reflect.Value
+	var err error
+
+	parser := jsonpath.New("sorting").AllowMissingKeys(true)
+	err = parser.Parse(t.field)
+	if err != nil {
+		glog.Fatalf("sorting error: %v\n", err)
+	}
+
+	iValues, err = findJSONPathResults(parser, iObj)
```

Comment:
This is going to be really expensive for very large sets (you're doing O(N) parses).  Please leave TODOs that you need to build an array that contains the resolved fields, then sort that.